android/src/main/java/com/rudderstack/android/internal/infrastructure/AppInstallUpdateTrackerPlugin.kt

```diff
@@ -0,0 +1,82 @@
+package com.rudderstack.android.internal.infrastructure
+
+import android.app.Application
+import com.rudderstack.android.androidStorage
+import com.rudderstack.android.currentConfigurationAndroid
+import com.rudderstack.android.storage.AndroidStorage
+import com.rudderstack.android.utilities.AppVersion
+import com.rudderstack.android.utilities.AppVersionManager
+import com.rudderstack.core.Analytics
+import com.rudderstack.core.InfrastructurePlugin
+import com.rudderstack.core.Logger
+
+class AppInstallUpdateTrackerPlugin : InfrastructurePlugin {
+
+    private var analytics: Analytics? = null
+    private lateinit var appVersion: AppVersion
+
+    companion object {
+        private const val PREVIOUS_VERSION = "previous_version"
+        private const val PREVIOUS_BUILD = "previous_build"
+        private const val VERSION = "version"
+        private const val BUILD = "build"
+
+        const val EVENT_NAME_APPLICATION_INSTALLED = "Application Installed"
+        const val EVENT_NAME_APPLICATION_UPDATED = "Application Updated"
+    }
+
+    override fun setup(analytics: Analytics) {
+        this.analytics = analytics
+        initialiseAppVersion(analytics.androidStorage, analytics.logger, analytics.currentConfigurationAndroid?.application)
+        if (this.analytics?.currentConfigurationAndroid?.trackLifecycleEvents == true) {
+            trackApplicationStatus()
+        }
+    }
+
+    private fun initialiseAppVersion(analyticsStorage: AndroidStorage, logger: Logger, application: Application?) {
+        val appVersionManager = AppVersionManager(application, analyticsStorage, logger)
```

Comment:
These two lines are ambiguous and not self explanatory. Use extension function for these.

Reply:
I've tried to improve this.